test-case/multiple-pause-resume.sh

```diff
@@ -0,0 +1,212 @@
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['t']='tplg'     OPT_DESC_lst['t']='tplg file, default value is env TPLG: $TPLG'
+OPT_PARM_lst['t']=1         OPT_VALUE_lst['t']="$TPLG"
+
+OPT_OPT_lst['l']='loop'     OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1         OPT_VALUE_lst['l']=3
+
+OPT_OPT_lst['c']='count'    OPT_DESC_lst['c']='test pipeline count'
+OPT_PARM_lst['c']=1         OPT_VALUE_lst['c']=2
+
+OPT_OPT_lst['r']='loop'     OPT_DESC_lst['r']='pause resume repeat count'
+OPT_PARM_lst['r']=1         OPT_VALUE_lst['r']=3
+
+OPT_OPT_lst['i']='min'      OPT_DESC_lst['i']='random range min value, unit is ms'
+OPT_PARM_lst['i']=1         OPT_VALUE_lst['i']='100'
+
+OPT_OPT_lst['a']='max'      OPT_DESC_lst['a']='random range max value, unit is ms'
+OPT_PARM_lst['a']=1         OPT_VALUE_lst['a']='200'
+
+OPT_OPT_lst['s']='sof-logger'   OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0             OPT_VALUE_lst['s']=1
+
+func_opt_parse_option $*
+
+repeat_count=${OPT_VALUE_lst['r']}
+loop_count=${OPT_VALUE_lst['l']}
+# configure random value range
+rnd_min=${OPT_VALUE_lst['i']}
+rnd_max=${OPT_VALUE_lst['a']}
+rnd_range=$[ $rnd_max - $rnd_min ]
+[[ $rnd_range -le 0 ]] && dlogw "Error random range scope [ min:$rnd_min - max:$rnd_max ]" && exit 2
+
+tplg=${OPT_VALUE_lst['t']}
+func_pipeline_export $tplg
+# get the min value of TPLG:'pipeline count' with Case:'pipeline count'
+[[ $PIPELINE_COUNT -gt ${OPT_VALUE_lst['c']} ]] && max_count=${OPT_VALUE_lst['c']} || max_count=$PIPELINE_COUNT
+[[ $max_count -eq 1 ]] && dlogw "pipeline count is 1, don't need to run this case" && exit 2
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+func_lib_setup_kernel_last_line
+
+declare -a pipeline_idx_lst
+declare -a cmd_idx_lst
+declare -a file_idx_lst
+
+# merge all pipeline to the 1 group
+for i in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+do
+    pipeline_idx_lst=(${pipeline_idx_lst[*]} $i)
+    type=$(func_pipeline_parse_value $i type)
+    if [ "$type" == "playback" ];then
+        cmd_idx_lst=(${cmd_idx_lst[*]} "aplay")
+        file_idx_lst=(${file_idx_lst[*]} "/dev/zero")
+    elif [ "$type" == "capture" ];then
+        cmd_idx_lst=(${cmd_idx_lst[*]} "arecord")
+        file_idx_lst=(${file_idx_lst[*]} "/dev/null")
+    elif [ "$type" == "both" ];then
+        cmd_idx_lst=(${cmd_idx_lst[*]} "aplay")
+        file_idx_lst=(${file_idx_lst[*]} "/dev/zero")
+        # both include playback & capture, so duplicate it
+        pipeline_idx_lst=(${pipeline_idx_lst[*]} $i)
+        cmd_idx_lst=(${cmd_idx_lst[*]} "arecord")
+        file_idx_lst=(${file_idx_lst[*]} "/dev/null")
+    else
+        dloge "Unknow pipeline type: $type"
+        exit 1
+    fi
+done
+pipeline_list_count=$(expr ${#pipeline_idx_lst[*]} - 1)
+# x here as the flag for pipeline split
+tmp_lst_str=$(echo ${pipeline_idx_lst[*]}|sed "s/ /x,/g")"x"
+for i in $(seq 1 $max_count)
```

Comment:
here base on how many pipeline need to do multiple  
so it will be `max_count` same with multiple-pipeline test-case  
for example: I have pipeline 1-4, I just want to combine 2 as test:  
it will be `(1 2)(1 3)(1 4)(2 3)(2 4)(3 4)`, but if this value is pipeline count
it will be `( 1 2 3 4)` ....

Reply:
I misunderstand the max_count meaning. OK, the code logic is right.